### PR TITLE
Minor updates to made package compatible with Python 3.11 and 3.12

### PR DESCRIPTION
**Pull request by Raalsky**
## Before submitting checklist

- [x] Did you **update the CHANGELOG**? (not for test updates, internal changes/refactors or CI/CD setup)
- [ ] Did you **ask the docs owner** to review all the user-facing changes?

### CODE REVIEW

**Comment by Raalsky** [comment]
E2E: https://github.com/neptune-ai/neptune-client/actions/runs/8222082577

**Review by szysad — APPROVED** [review]
Big part of this PR is syntax change from
```
def f():
    ...
```
to
```
def f(): ...
```
Can you just explain a big how why is this necessary?

Besides that looks solid 🪨🤘

File: .github/workflows/e2e.yml
[the remark this answers is not in the document]
Comment (Raalsky) [reply .github/workflows/e2e.yml:1]: We used to test with the latest supported and the oldest one. So it still remains by testing Python 3.12 instead.

File: src/neptune/attributes/atoms/copiable_atom.py
[the remark this answers is not in the document]
Comment (Raalsky) [reply src/neptune/attributes/atoms/copiable_atom.py:1]: Just `black` :P. You can check commit by commit as there is one about `black` suggestions.

**Pull request merged**